CMakeLists.txt

```
@@ -205,6 +206,18 @@ else()
     endif()
 endif()
 
+if(UR_ENABLE_EXCEPTION_SANITIZER)
+    if(APPLE)
+        message(WARNING "Exception sanitizer layer isn't supported on macOS")
+        set(UR_ENABLE_EXCEPTION_SANITIZER OFF)
+    elseif(WIN32)
+        message(WARNING "Exception sanitizer layer isn't supported on Windows")
+        set(UR_ENABLE_EXCEPTION_SANITIZER OFF)
+    else()
+        add_compile_definitions(UR_ENABLE_EXCEPTION_SANITIZER)
+    endif()
+endif()
```

Always enable building the layer on Linux when developer mode is enabled.

```suggestion
if(UR_DEVELOPER_MODE AND CMAKE_SYSTEM_NAME STREQUAL Linux)
    set(UR_ENABLE_EXCEPTION_SANITIZER ON)
    add_compile_definitions(UR_ENABLE_EXCEPTION_SANITIZER)
endif()
```